### PR TITLE
Fix: react-query typescript in connect-query-plugin

### PR DESCRIPTION
**Pull request by vctqs1**
I was discovered that there is an inconsistency with the parameters passed to the useInfiniteQuery function, particularly when using use<Name>InfiniteQuery. This inconsistency results in type mismatch errors.

Current Implementation:

The useInfiniteQuery function has the following signature:

```ts
useInfiniteQuery: <ParamKey extends keyof PartialMessage<I>>(
  input: DisableQuery | PartialMessage<I>,
  options: BaseInfiniteQueryOptions<I, O, ParamKey> & RequireExactlyOne<{
    applyPageParam: (options: {
      pageParam: PartialMessage<I>[ParamKey] | undefined;
      input: PartialMessage<I>;
    }) => PartialMessage<I>;
    pageParamKey: ParamKey;
  }>
) => {
  // Implementation details...
};

```

However, in the useSayInfiniteQuery function, the queryOptions and options are made optional:
```
export const useSayInfiniteQuery = (
  inputs: Parameters<typeof say.useInfiniteQuery>[0],
  queryOptions?: Partial<UseInfiniteQueryOptions<PartialMessage<SayRequest>, ConnectError>>,
  options?: Parameters<typeof say.useInfiniteQuery>[1],
) => {
  const baseOptions = say.useInfiniteQuery(inputs, options);
  return useInfiniteQuery({
    ...baseOptions,
    ...queryOptions,
  });
};
```
However, the `options` parameter is currently required in `useInfiniteQuery`. To address this, we propose swapping the order of queryOptions and options, making `options` required instead. This adjustment will not only ensure consistency but also enhance code readability. We'll also need to update the order of parameters for other hooks to maintain uniformity across the project

**PLEASE HELP HAVE A LOOK**
Also, there is an issue with the useInfiniteQuery function when the InputRequest is an empty object (PartialMessage<I> = {}). This leads to a TypeScript error in certain scenarios.

```
useInfiniteQuery: <ParamKey extends keyof PartialMessage<I>>(input: DisableQuery | PartialMessage<I>, options: BaseInfiniteQueryOptions<I, O, ParamKey> & RequireExactlyOne<{
        applyPageParam: (options: {
            pageParam: PartialMessage<I>[ParamKey] | undefined;
            input: PartialMessage<I>;
        }) => PartialMessage<I>;
        pageParamKey: ParamKey;
    }>) => {
        enabled: boolean;
        queryKey: ConnectQueryKey<I>;
        queryFn: (context: QueryFunctionContext<ConnectQueryKey<I>, PartialMessage<I>[ParamKey]>) => Promise<O>;
        getNextPageParam: GetNextPageParamFunction<O>;
        onError?: (error: ConnectError) => void;
    };
```
with
- PartialMessage<I> = {}
- <ParamKey extends keyof PartialMessage<I>> = never

EX in `useWorkInfiniteQuery`
it said 
```
      Types of property 'queryFn' are incompatible.
        Type '(context: QueryFunctionContext<ConnectQueryKey<Nothing>, never>) => Promise<Nothing>' is not assignable to type 'QueryFunction<Nothing, ConnectQueryKey<Nothing>>'.
```
also tried to rteplace ConnectQueryKey = never but it's said 
```
        Type '(context: QueryFunctionContext<ConnectQueryKey<Nothing>, never>) => Promise<Nothing>' is not assignable to type 'QueryFunction<Nothing, ConnectQueryKey<Nothing>>'.

```
image
I've make an change like this
```
return useInfiniteQuery<Nothing, ConnectError, Nothing, keyof typeof inputs extends never ? any : ConnectQueryKey<Nothing>>({
        ...baseOptions,
        ...queryOptions,
    });
```

if keyof InputRquest is `never`, take `any` to bypass this situation.

Otherwise, it works well in others
image

### CODE REVIEW

**Comment by paul-sachs** [comment]
@vctqs1 Things just got a little busy on my end, but I promise I haven't forgotten. I'll take a look at this sometime this week, just trying to clear my plate a little.

File: packages/generated-react/buf.gen.yaml [hunk 1]
```diff
@@ -25,5 +25,6 @@ plugins:
     path: ../protoc-gen-connect-query-react/bin/protoc-gen-connect-query-react
     opt:
       - target=ts
+      - import_extension=none
```
Comment (paul-sachs) [inline packages/generated-react/buf.gen.yaml:28]: Any specific reason we changed this? It's not a big deal, but i'd like to make sure we're only making the necessary changes.

Comment (paul-sachs) [reply packages/generated-react/buf.gen.yaml:28]: And since no other plugins have this enabled, i'd rather we be consistent.

Comment (vctqs1) [reply packages/generated-react/buf.gen.yaml:28]: > And since no other plugins have this enabled, i'd rather we be consistent.

Ah let me change it back

Comment (vctqs1) [reply packages/generated-react/buf.gen.yaml]: I just updated it, thank you @paul-sachs

**Review by paul-sachs — APPROVED** [review]

**Comment by vctqs1** [comment]
@paul-sachs hi,  just resolve the conflicts, please help me review it when you can. thank you

**Review by paul-sachs — APPROVED** [review]

**Comment by vctqs1** [comment]
@paul-sachs please help me know if I need to do anything else to get it merge thank you

**Comment by vctqs1** [comment]
@paul-sachs Could you help, please let me know when it can be published? I'm waiting to use it

**Pull request merged**